addons/account/models/account_move.py

```diff
@@ -3230,6 +3223,55 @@ def _get_new_hash(self, secure_seq_number):
         #build and return the hash
         return self._compute_hash(prev_move.inalterable_hash if prev_move else u'')
 
+    def _hash_moves(self, force=False):
+        self.flush_recordset()  # Ensure that the name is correctly computed before it is used to generate the hash
+        for move in self._get_moves_to_hash(force=force):
+            new_number = move.journal_id.secure_sequence_id.next_by_id()
+            super(AccountMove, move).write({
```

Comment:
It's a bit weird to call `super().write` outside of `write`. It might have weird behaviors the `_hash_moves` is called outside of `write`
```suggestion
            super().write({
```

Reply:
It isn't working, nor the simple assignment move.inalterable_hash = ... :thinking:

Reply:
It's not expected to work indeed if you call it in `write`, but it's still not an usual pattern (it's not wrong per se)
But removing the arguments of `super` should work :wink:

Reply:
Removing move in the super call makes all moves in self have the same secure_sequence_number of the first move ://

Reply:
Indeed :facepalm: